src/drawers/lineDrawer.ts

```diff
@@ -5,56 +5,40 @@
 
 import * as d3 from "d3";
 
-import { Dataset } from "../core/dataset";
+import { AttributeToAppliedProjector, SimpleSelection } from "../core/interfaces";
+import { CanvasDrawStep } from "./canvasDrawer";
+import { SVGDrawer } from "./svgDrawer";
 
-import { SimpleSelection } from "../core/interfaces";
-import { Drawer } from "./drawer";
-import { AppliedDrawStep } from "./index";
-
-export class Line extends Drawer {
-  private _d3LineFactory: () => d3.Line<any>;
-
-  /**
-   * @param dataset
-   * @param _d3LineFactory A callback that gives this Line Drawer a d3.Line object which will be
-   * used to draw with.
-   */
-  constructor(dataset: Dataset, d3LineFactory: () => d3.Line<any>) {
-    super(dataset);
-    this._d3LineFactory = d3LineFactory;
-    this._className = "line";
-    this._svgElementName = "path";
+export class LineSVGDrawer extends SVGDrawer {
+  constructor() {
+    super("path", "line");
   }
 
   protected _applyDefaultAttributes(selection: SimpleSelection<any>) {
-    super._applyDefaultAttributes(selection);
     selection.style("fill", "none");
   }
 
   public selectionForIndex(index: number): SimpleSelection<any> {
     return d3.select(this.selection().node());
   }
+}
 
-  /**
-   *
-   * @param data Data to draw. The data will be passed through the line factory in order to get applied
-   * onto the canvas.
-   * @param step
-   * @private
-   */
-  protected _drawStepCanvas(data: any[][], step: AppliedDrawStep) {
-    const context = this.canvas().node().getContext("2d");
-
-    const d3Line = this._d3LineFactory();
-
-    const attrToAppliedProjector = step.attrToAppliedProjector;
+/**
+ * @param d3LineFactory A callback that gives this Line Drawer a d3.Line object which will be
+ * used to draw with.
+ *
+ * TODO put the d3.Line into the attrToAppliedProjector directly
+ */
+export function makeLineCanvasDrawStep(d3LineFactory: () => d3.Line<any>): CanvasDrawStep {
+  return (context: CanvasRenderingContext2D, data: any[][], attrToAppliedProjector: AttributeToAppliedProjector) => {
+    const d3Line = d3LineFactory();
     const resolvedAttrs = Object.keys(attrToAppliedProjector).reduce((obj, attrName) => {
-      obj[attrName] = attrToAppliedProjector[attrName](data, 0);
+      if (attrName !== "d") {
```

Comment:
Yep, eventually we should make the LineSVGDrawer put the "d" attribute, instead of the LinePlot creating the "d" attribute